onnxruntime/core/graph/graph.cc

```diff
@@ -467,6 +480,188 @@ void Node::ToProto(NodeProto& proto, bool update_subgraphs) const {
+  auto LoadNodeArgsOrtFormat =
+      [&](const flatbuffers::Vector<flatbuffers::Offset<flatbuffers::String>>* fbs_node_arg_names,
+          std::vector<NodeArg*>& node_args,
+          bool check_parent_graph = false) -> Status {
+    ORT_RETURN_IF(nullptr == fbs_node_arg_names, "fbs_node_arg_names cannot be null");
+    node_args.reserve(fbs_node_arg_names->size());
+    if (fbs_node_arg_names) {
+      for (const auto& node_arg_name : *fbs_node_arg_names) {
+        ORT_RETURN_IF(nullptr == node_arg_name, "node_arg_name cannot be null");
+        auto* node_arg = check_parent_graph ? graph_->GetNodeArgIncludingParentGraphs(node_arg_name->str())
+                                            : graph_->GetNodeArg(node_arg_name->str());
+        ORT_RETURN_IF(nullptr == node_arg, "Could not find NodeArg ", node_arg_name->str());
+        node_args.push_back(node_arg);
+      }
+    }
+    return Status::OK();
+  };
+
+  FBS_SET_STR_VAL(name_, fbs_node.name());
+  FBS_SET_STR_VAL(description_, fbs_node.doc_string());
+  FBS_SET_STR_VAL(domain_, fbs_node.domain());
+  since_version_ = fbs_node.since_version();
+  FBS_SET_STR_VAL(op_type_, fbs_node.op_type());  // index was set in ctor
+  node_type_ = static_cast<Node::Type>(fbs_node.type());
+  FBS_SET_STR_VAL(execution_provider_type_, fbs_node.execution_provider_type());
+  ORT_RETURN_IF_ERROR(LoadNodeArgsOrtFormat(fbs_node.inputs(), definitions_.input_defs));
+
+  // attributes
+  auto fbs_attributes = fbs_node.attributes();
+  if (fbs_attributes) {
+    for (const auto* fbs_attr : *fbs_attributes) {
+      ORT_RETURN_IF(nullptr == fbs_attr, "fbs_attr cannot be null");
+      AttributeProto attr_proto;
+      std::unique_ptr<Graph> subgraph;
+      ORT_RETURN_IF_ERROR(
+          experimental::utils::LoadAttributeOrtFormat(*fbs_attr, attr_proto, subgraph, *graph_, *this, logger));
+
+      // If we have a sub graph in this attributes, it will be loaded into subgraph ptr
+      // while the attribute proto contains the sub graph will have the empty g() field
+      if (attr_proto.type() == AttributeProto_AttributeType_GRAPH) {
+        ORT_RETURN_IF_NOT(subgraph, "Serialization error. Graph attribute was serialized without Graph instance");
+        attr_to_subgraph_map_.emplace(attr_proto.name(), gsl::not_null<Graph*>(subgraph.get()));
+        subgraphs_.push_back(std::move(subgraph));
+      }
+
+      AddAttribute(attr_proto.name(), attr_proto);
+    }
+  }
+
+  ORT_RETURN_IF_ERROR(LoadNodeArgsOrtFormat(fbs_node.implicit_inputs(), definitions_.implicit_input_defs,
+                                            /* check parent graphs */ true));
+
+  {  // input_arg_counts
+    auto fbs_input_arg_counts = fbs_node.input_arg_counts();
+    ORT_RETURN_IF(nullptr == fbs_input_arg_counts, "fbs_input_arg_counts cannot be null");
+    auto& input_arg_count = definitions_.input_arg_count;
```

Comment:
general nit: an error message like this isn't that helpful to a user or someone not familiar with the serialization as it just says what is wrong not why. prefer something that includes the latter. in this case something like 'Serialization error. Node input arg counts is missing' is more helpful as it provides a hint as to where things probably went wrong along with some context (serialization was involved as was a Node).